sql/planbuilder/cte.go

```diff
@@ -188,28 +206,28 @@ func (b *Builder) buildRecursiveCte(inScope *scope, union *ast.Union, name strin
 // "should have one or more non-recursive query blocks followed by one or more recursive ones"
 // "the recursive table must be referenced only once, and not in any subquery"
 func splitRecursiveCteUnion(name string, n ast.SelectStatement) (ast.SelectStatement, ast.SelectStatement) {
-	union, ok := n.(*ast.Union)
```

Comment:
I think the variable in this function should still be called `union`. After all, we know it's a union.